### ISSUE TITLE
7.72 Incoherence on OTBM load map versions.

### ISSUE THREAD

**Issue by danilopucci** [nekiro/TFS-1.5-Downgrades#118]
**Protocol version**
7.72

**Describe the bug**
Incoherence on OTBM load map versions.

**To Reproduce**
[iomap.cpp, near line 90](https://github.com/nekiro/TFS-1.5-Downgrades/blob/7.72/src/iomap.cpp#L90):
```cpp		
uint32_t headerVersion = root_header.version;
/*if (headerVersion == 0) {
    //In otbm version 1 the count variable after splashes/fluidcontainers and stackables
    //are saved as attributes instead, this solves a lot of problems with items
    //that are changed (stackable/charges/fluidcontainer/splash) during an update.
    setLastErrorString("This map need to be upgraded by using the latest map editor version to be able to load correctly.");
    return false;
 }

    WARNING: I had to comment that to make the server load my 7.6 rme generated map, read the note ^.
*/
```
The RME generated map hosted on this branch has headerVersion = 2. The above commented block would be right, if there is a change on other part of code, which includes the read the count variable after splashes/fluidocontainers and stackables.

[item.cpp, near line 144](https://github.com/nekiro/TFS-1.5-Downgrades/blob/7.72/src/item.cpp#L144)
```cpp
    unsigned char count = 0;
    if(iType.stackable || iType.isSplash() || iType.isFluidContainer()){
        if(!propStream.read<uint8_t>(count)){
                    return nullptr;
        }
    }

    return Item::CreateItem(id, count);
```
This way the real 7.6 map gerenated by RME would load correctly(with the older way to load count variable) and the commented block would be fine to be as it is.

The way that the code is now at this branch, it is not possible to load a real 7.6 map correcly.

My opinion is to make it dinamically, letting both maps be loaded, [same way as Remeres did it](https://github.com/hampusborgos/rme/blob/master/source/iomap_otbm.cpp#L76)

And finally, to make it fully compatible, a small modification on OTBI load:
[items.cpp, near line 436](https://github.com/nekiro/TFS-1.5-Downgrades/blob/7.72/src/items.cpp#L436)
```cpp
		switch (itemNode.type) {
			case ITEM_GROUP_CONTAINER:
				iType.type = ITEM_TYPE_CONTAINER;
				break;
			case ITEM_GROUP_DOOR:
				//not used
				iType.type = ITEM_TYPE_DOOR;
				break;
			case ITEM_GROUP_MAGICFIELD:
				//not used
				iType.type = ITEM_TYPE_MAGICFIELD;
				break;
			case ITEM_GROUP_TELEPORT:
				//not used
				iType.type = ITEM_TYPE_TELEPORT;
				break;
			case ITEM_GROUP_CHARGES:
			        iType.type = ITEM_TYPE_RUNE;
			        break;
			case ITEM_GROUP_NONE:
			case ITEM_GROUP_GROUND:
			case ITEM_GROUP_SPLASH:
			case ITEM_GROUP_FLUID:
			case ITEM_GROUP_DEPRECATED:
			case ITEM_GROUP_AMMUNITION:
			case ITEM_GROUP_ARMOR:
			case ITEM_GROUP_WEAPON:
			case ITEM_GROUP_WRITEABLE:
			case ITEM_GROUP_KEY:
			       break;
			default:
				return false;
		}
```
The modification above is to avoid returning false on flags of items.otb version 1 and 2, leading a return false on "default" statement.
This modification is simple adding some ITEM_GROUP enums values to the switch-case

**Comment by nekiro** [comment]
Feel free to open a PR. I don't consider this a must have.

**Comment by danilopucci** [comment]
solved on PR #119
